presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -693,4 +697,38 @@ public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @S
 
         return lower;
     }
+
+    @Description("Cosine similarity between two Map<Varchar, Double>")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cosineSimilarity(@SqlType(MAP_STRING_DOUBLE) Block leftMap, @SqlType(MAP_STRING_DOUBLE) Block rightMap)
```

Comment:
Having this operate on `map(varchar, double)` seems strange to me. cosine similarity is generally defined for vectors which typically are indexed by natural numbers. I think we should make this work for any type K which is comparable